File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        pos=[]
        neg = []

        for i in nums:
            if i<0:
                neg.append(i)

            if i>=0:
                pos.append(i)


        ans = [0]*len(nums)
        # ans[0::2] = pos
        # ans[1::2] = neg

        for i in range(len(nums)):
            if i%2 == 0:
                ans[i] = pos[i//2]

            else:
                ans[i] = neg[i//2]

        return ans
        
            

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> pos;
        vector<int> neg;

        for(auto x: nums){
            if(x>0) pos.push_back(x);
            else neg.push_back(x);
        }

        int n = nums.size();

        vector<int> ans(n, 0);

        for(int i = 0; i<n; i++){
            if(i%2 == 0) ans[i] = pos[i/2];
            else ans[i] = neg[i/2];
        }

        return ans;
    }
};